src/components/timecard/edit-shift-hours/EditShiftHours.jsx

```diff
@@ -59,20 +62,37 @@ const EditShiftHours = ({
     }
   };
 
-  const handleServerValidationErrors = (error) => {
+  const handleServerValidationErrors = (errors) => {
+    if (errors === null || !Array.isArray(errors)) {
+      return false;
+    }
     const summaryErrors = {};
     let errorsHandled = true;
 
-    if (
-      error ==
-      ' has the following error(s): Time periods must not overlap with another time period'
-    ) {
-      summaryErrors['shift-start-time'] = {
-        message: 'Time periods must not overlap with another time period',
-      };
-    } else {
-      errorsHandled = false;
-    }
+    errors.forEach((error) => {
+      if (error.field === 'startAndEndTime') {
+        setClashingTimes(error.data);
+        setClashingProperty(error.field);
+        summaryErrors['shift-start-time'] = {
+          message:
+            'Your start and finish times must not overlap with another time period',
+        };
+      } else if (error.field === 'startTime') {
+        setClashingTimes(error.data);
+        setClashingProperty(error.field);
+        summaryErrors['shift-start-time'] = {
+          message: 'Your start time must not overlap with another time period',
+        };
+      } else if (error.field === 'endTime') {
+        setClashingTimes(error.data);
+        setClashingProperty(error.field);
+        summaryErrors['shift-finish-time'] = {
+          message: 'Your end time must not overlap with another time period',
+        };
+      } else {
+        errorsHandled = false;
```

Comment:
We only hit this if the server has given us some dodgy data e.g a clashing entry without an expected error.field right..
If so, I think that means we can break the loop here as we're just going to show the service error.

Reply:
I've made this change, do you think this function is harder to read now, with just the boolean returns and no variable name to indicate what they mean, or is it still ok to follow?

Reply:
I don't think it's become particularly more difficult to follow. 
Buttt if you wanted to maintain that extra level of readability, you could use a `for of `loop instead and then `break` instead of return false.
E.g. (untested)
```
for (const error of errors) {
      if (error.field === 'startAndEndTime') {
        setClashData(
          'shift-start-time',
          'Your start and finish times must not overlap with another time period'
        );
      } else if (error.field === 'startTime') {
        setClashData(
          'shift-start-time',
          'Your start time must not overlap with another time period'
        );
      } else if (error.field === 'endTime') {
        setClashData(
          'shift-finish-time',
          'Your finish time must not overlap with another time period'
        );
      } else {
        errorsHandled = false;
        break;
      }
    }
```

Reply:
I like this, have made the change now :)